//Add which syntax is used, proto2 by default
syntax = 'proto3';

package blog;

//Define the service
service BlogPostService {
	//Add methods to the services
	//Here the communication type is the simplest 'Unary rpc'
	rpc GetOne (GetOneRequest) returns (GetOneResponse);
}

//Describe the methods data structure
message GetOneRequest {
	int32 post_id = 1;
}

message GetOneResponse {
    BlogPost post = 1;
	Author author = 2;
}

message BlogPost {
	string title = 1;
	string description = 2;
	string content = 3;
    repeated string tags = 4;
}

message Author {
	string first_name = 1;
	string last_name = 2;
}
